app/services/catalog/update_order_item.rb

```diff
@@ -0,0 +1,33 @@
+module Catalog
+  class UpdateOrderItem
+    class OrderItemNotFound < StandardError; end
+
+    def initialize(topic)
```

Comment:
@eclarizio @syncrou Should we pass in the whole topic here, is it a topic or is it an event/message.

Reply:
I originally called it topic on the approval_listener to match the method name `subscribe_topic`. Talking to Bill, the term 'topic', or 'event' both mean the same thing.  

If you just pass the payload though you won't be able to see the message ( `topic.message` ) as the message is not part of the `payload` key. 

So the 'topic' seems more flexible to pass in from my perspective.